backend/server/report_store.py

```
@@ -0,0 +1,57 @@
+
+    async def _ensure_parent_dir(self) -> None:
+        self._path.parent.mkdir(parents=True, exist_ok=True)
+
+    async def _read_all_unlocked(self) -> Dict[str, Dict[str, Any]]:
+        if not self._path.exists():
+            return {}
+        try:
+            data = json.loads(self._path.read_text(encoding="utf-8"))
+            if isinstance(data, dict):
+                return data  # type: ignore[return-value]
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)